pkg/gzip/gzip.go

```diff
@@ -165,3 +191,11 @@ func ExtractTarGz(owner string, fp io.Reader, dest string) (err error) {
 	}
 	return nil
 }
+
+func firstExistingParent(path string) string {
+	for _, err := os.Stat(path); os.IsNotExist(err); {
+		path = filepath.Dir(path)
+		_, err = os.Stat(path)
```

Comment:
what if you got an error that was not `os.IsNotExist`? you would just return a garbage path right?